\abstract{
\textbf{Motivation:}\
The advent of modern high-throughput genetics has broadened the gap between the ever-growing volume of sequencing data against the tools required to process them. The need to pinpoint a small subset of functionally important variants has now shifted towards identifying the critical differences between normal variants and disease-causing ones.\
\
Our sequence analysis pipeline (\app) is an offline open-source multi-step analysis environment designed to annotate and extract useful variants from Variant Call Format (VCF) files under an inheritance context though top-down filtering via swappable modules run entirely off a live bootable medium accessed locally through a web-browser.\\
\
\textbf{Methods:}\
The pipeline consists of three key stages that pertain to the separate modes of annotation, filtering, and interpretation. Core annotation performs variant-mapping to gene-isoforms at the exon/intron level, append functional data pertaining the type of variant mutation, and determine hetero/homozygosity. Up to 12 filtering modules can be used in sequence ranging from single quality control to multi-file penetrance model specifics such as X-linked recessive or mosaicism. Depending on the type of interpretation required, additional annotation is performed to identify organ specificity through gene expression and protein domains.\\
\
\textbf{Results:}\
\
Three case studies were analysed; 2 of autosomal recessive inheritance, and an X-linked dominant. \app made effective use of different module configurations to identify singular rare causative variants from whole-exome sequenced input samples.
\\
\textbf{Availability:} \app is licensed under GPLv3 and is hosted and maintained via Bitbucket.\\\
\
\textit{Source Code and Live ISO:}\ \ \url|https://www.bitbucket.io/momo13/ovas-pipeline.git|\\
\
\textbf{Supplementary information:} Supplementary data is available from \textit{Bioinformatics} online.\\
\textbf{Contact:} \href{r.kleta@ucl.ac.uk}{r.kleta@ucl.ac.uk}\
}
