
%% Results table

\begin{table}[!b]
\processtable{Average single-core runtimes of VCF files containing 50,000 variants passing individually through all filters\label{table:results}}
{\begin{tabular}{| c | *2c |} \hline %\toprule
%\emph{Pipeline} & \emph{Module} & \emph{Runtime} \\
%\emph{Stage} & \emph{Name} & \emph{(seconds)} \\
\emph{Pipeline Stage} & \emph{Module Name} & \emph{Runtime (seconds)} \\
\hline
% A multirow would be more benefial here
% but let's keep package req down for now
             & Gene     & 179\\
Annotation   & Function & 28.7\\
             & Zygosity & 0.81\\
\hline
             & Physical Location & 1.02 \\
             & Read Depth        & 1.26 \\
             & Call Quality      & 0.93\\
Filtering    & AAF               & 432\\
             & Mutation Type     & 1.08\\
             & Novel Variant     & 1.12\\
             & Common Gene       & 22.5\\
             & Common Variant    & 26.1\\
\hline
             & AD Inheritance    & 0.83\\
 Trait       & AR Inheritance    & 1.22\\
Penetrance   & XD Inheritance    & 0.74\\
  Model      & XR Inheritance    & 1.39\\
             & Mosaicism         & 0.94\\
\hline
                 & Isoform Context      & 2.28\\
   Extended      & Protein Context      & 4.10\\
 Annotation      & Gene Expression      & 145\\
                 & Transcription Factor & 0.86\\
                 & House-Keeping        & 0.93\\
\hline
%\botrule
\end{tabular}}
{Single VCF file timings for Annotation,Filtering, and Extended Annotation modules. Trait Penetrance module timings are based on three VCFs consisting of a parent-offspring trio. Tests were run on a 2GHz dual-core processor with 4GB RAM.}
\end{table}
