\section{Conclusion}

The self-contained environment provided by \app allows researchers to tailor all aspects of their analysis and retain control of their data sets at any phase of processing by means of the transparent open-source modules that comprise the pipeline. 

The live environment, paired with the web front-end, provides the additional advantage of abstracting the end-user from the underlying platform specifics by streamlining the input and configuration process, as well as logging active progress descriptions for the current stage of processing, and lastly providing a malleable final report upon all remaining variants discovered complete with dynamic filtering capabilities.

The entirety of all uploaded variants are processed first at the gene annotation stage, placing significant strain at the initial stage of the pipeline that is only managed through the use of employing C++ binaries to overcome the performance bottleneck that would otherwise exist with Python/Bash scripts.

The annotation step is crucial however, where the vast majority of whole-genome sequenced variants are deemed wholly intergenic and are filtered out as uninformative to the analysis. More common exome-sequencing data observe less of a reduction at a much faster processing rate due to the smaller number of total variants, but at the impediment of missing regulatory elements due to lack of coverage.




Future: pedigree creation, integrate haplo. File 
Conc: Future work would be to provide file snapshots at each stage of processing for better progress reporting, automatic pedigree-creation to 

parelliizaiotn

genemaps that dont exist, updating snp and fasta, requires internet retrieval.

reruns are trivial if annotation is done already